src/observations/process_radec.jl

```diff
@@ -44,7 +44,7 @@ function evaluate(res::AbstractVector{OpticalResidual{T, TaylorN{T}}}, x::Vector
     end
     return res_new
 end
-(res::AbstractVector{OpticalResidual{T, TaylorN{T}}})(x::Vector{T}) where {T <: Real} = evaluate(res, x)
+(res::Vector{OpticalResidual{T, TaylorN{T}}})(x::Vector{T}) where {T <: Real} = evaluate(res, x)
```

Comment:
Naive question: what is the motivation for this change? (I think `AbstractVector` would allow applying this function to more types, e.g. `SubArray{T,1}` or `SVector`).

Reply:
This may perhaps not be relevant, but I thought it was easier to ask...

Reply:
This helps avoid an ambiguity with callable methods from Interpolations.jl which also dispatch on `<:AbstractArray{T,N}`; I agree it's perhaps a bit restrictive, and a more generic solution could be to define a `OpticalResidualVector` struct and define the callable methods over that struct. I would suggest to go with this approach for now, and revisit this if needed.

Reply:
Yet another possibility is to avoid using `Vector{T}` for the calling argument type (again, maybe wrapping it somehow in another struct)